gensound/transforms.py

```diff
@@ -190,31 +190,50 @@ def realise(self, audio):
 class Fade(Transform):
     min_fade = -50
     
-    def __init__(self, is_in=True, duration=3e3):
-        self.is_in = is_in
+    def __init__(self, duration=None, curve=None, is_in=True, degree=None):
+        assert curve in (None, "linear", "polynomial")
+        # default setting for backwards compatibility (allowing usage of both Fade() and FadeIn/FadeOut()
+        if curve == None:
+            curve = "linear"
+        if duration == None:
+            duration = 3e3
+        if degree == None:
+            degree = 1
+        
+        self.curve = curve
         self.duration = duration
-    
+        self.is_in = is_in
+        self.degree = degree
+
     def realise(self, audio):
-        amp = DB_to_Linear(np.linspace(Fade.min_fade, 0, self.num_samples(audio.sample_rate)))
-        #amp = np.linspace(0, 1, self.num_samples(audio.sample_rate))
+        #curve type handler
+        if self.curve == "linear": amp = np.linspace(0, 1, self.num_samples(audio.sample_rate))
+        if self.curve == "polynomial": amp = (np.linspace(0, 1, self.num_samples(audio.sample_rate))) ** self.degree      
+        
```

Comment:
all of these `if`s deserve a line break, I don't see a reason to squish it all into one line.
Also, when using `if`s in succession, use `elif` instead